src/Compilers/CSharp/Portable/BoundTree/BoundTreeWalker.cs

```diff
@@ -142,6 +140,44 @@ protected virtual void BeforeVisitingSkippedBoundBinaryOperatorChildren(BoundBin
         {
         }
 
+        public sealed override BoundNode? VisitBinaryPattern(BoundBinaryPattern node)
+        {
+            if (node.Left is not BoundBinaryPattern binary)
+            {
+                return base.VisitBinaryPattern(node);
+            }
+
+            var rightOperands = ArrayBuilder<BoundPattern>.GetInstance();
+
+            rightOperands.Push(node.Right);
+            BeforeVisitingSkippedBoundBinaryPatternChildren(binary);
+            rightOperands.Push(binary.Right);
+
+            BoundPattern? current = binary.Left;
+
+            while (current.Kind == BoundKind.BinaryPattern)
+            {
+                binary = (BoundBinaryPattern)current;
+                BeforeVisitingSkippedBoundBinaryPatternChildren(binary);
+                rightOperands.Push(binary.Right);
+                current = binary.Left;
+            }
+
+            Visit(current);
+
+            while (rightOperands.TryPop(out current))
+            {
+                Visit(current);
+            }
+
+            rightOperands.Free();
+            return null;
+        }
+
+        protected virtual void BeforeVisitingSkippedBoundBinaryPatternChildren(BoundBinaryPattern node)
```

Comment:
>BeforeVisitingSkippedBoundBinaryPatternChildren [](http://example.com/codeflow?start=31&length=47)

It doesn't look like this method is overridden by anyone. I think we should remove it for now. It can be added later when needed.